R/busca_multi.R

```diff
@@ -0,0 +1,20 @@
+#' @title Busca por CEP
+#' @description Busca usando multiplos CEP (search for information using multiple postal code).
+#' @importFrom magrittr %>%
+#' @importFrom magrittr %<>%
+#' @importFrom dplyr add_row
+#' @param lista_ceps lista de CEPs (postal code list).
+#' @param token Token de autorização. Veja <http://cepaberto.com/users/register>.
+#' @export
+
+busca_multi <- function( lista_ceps = c("01001000", "01001001") , token = NULL ) {
+  CEPS <- parse_api(NA) %>% na.exclude()
+  lista_ceps %<>%  as.vector()
+  for (i in lista_ceps) {
```

Comment:
Also getting this when running `busca_multi()`:

```r
> busca_multi(token = token)
Error: Can't combine `..1$ddd` <character> and `..2$ddd` <integer>.
Run `rlang::last_error()` to see where the error occurred.
> rlang::last_error()
<error/vctrs_error_incompatible_type>
Can't combine `..1$ddd` <character> and `..2$ddd` <integer>.
Backtrace:
  1. global::busca_multi(token = token)
  2. [ `%<>%`(...) ] with 7 more calls
 10. tibble::add_row(., busca_cep(cep = i, token = token))
 11. tibble:::rbind_at(.data, df, pos)
 12. vctrs::vec_rbind(old, new)
 14. vctrs::vec_default_ptype2(...)
 15. vctrs::stop_incompatible_type(...)
 16. vctrs:::stop_incompatible(...)
 17. vctrs:::stop_vctrs(...)
Run `rlang::last_trace()` to see the full context.
> rlang::last_trace()
<error/vctrs_error_incompatible_type>
Can't combine `..1$ddd` <character> and `..2$ddd` <integer>.
Backtrace:
     █
  1. └─global::busca_multi(token = token)
  2.   └─CEPS %<>% add_row(busca_cep(cep = i, token = token))
  3.     ├─base::withVisible(eval(quote(`_fseq`(`_lhs`)), env, env))
  4.     └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
  5.       └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
  6.         └─`_fseq`(`_lhs`)
  7.           └─magrittr::freduce(value, `_function_list`)
  8.             ├─base::withVisible(function_list[[k]](value))
  9.             └─function_list[[k]](value)
 10.               └─tibble::add_row(., busca_cep(cep = i, token = token))
 11.                 └─tibble:::rbind_at(.data, df, pos)
 12.                   └─vctrs::vec_rbind(old, new)
 13.                     └─(function () ...
 14.                       └─vctrs::vec_default_ptype2(...)
 15.                         └─vctrs::stop_incompatible_type(...)
 16.                           └─vctrs:::stop_incompatible(...)
 17.                             └─vctrs:::stop_vctrs(...)
```

Reply:
Regarding the error:
The changes I proposed to parse_api() - including specification of a format for possible NAs - were supposed to address it without changing any functionality of parse_api(). When running it from my end with the updated parse_api it works just fine.
![image](https://user-images.githubusercontent.com/55462632/93602519-15d7d400-f999-11ea-9154-7d289386a339.png)

the line 
```
CEPS <- parse_api(NA) %>% na.exclude()
```
Is indeed bothering me, but it was the simplest way to initialize an empty tibble for the loop I managed to write. I am not sure if there is a better way to do it, but this works fine and considering the mandatory 1 second pauses in between searches I was not too worried about minimal performance hits.

Reply:
Yeah, it works fine for me now, the mistake was on my end. Regarding `na.exclude()`, I was wondering initially why you would do it that way but it makes perfect sense. I'm just tidying up some small things in the merge and I'll submit the new version to CRAN asap. Thanks for your contribution, Rodrigo!